src/App.js

```diff
@@ -23,35 +23,35 @@ const App = () => {
     const y = 0;
     const newSolarPanel = { x: x, y: y, length: length, width: width, powerCapacity: powerCapacity, type: 'solar' };
     setSolarPanels([...solarPanels, newSolarPanel]);
-    setPowerOutput(powerOutput + powerCapacity);
+    
   };
 
   const handleSolarPanelRemove = (index) => {
     const removedSolarPanel = solarPanels[index];
     setSolarPanels(solarPanels.filter((_, i) => i !== index));
-    setPowerOutput(powerOutput - (isNaN(removedSolarPanel.powerCapacity) ? 0 : removedSolarPanel.powerCapacity ));
+ 
   };
 
-  const handleSolarPanelUpdate = (index, length, width, powerCapacity) => {
+  const calculateTotalCapacity = () => {
```

Comment:
Is there a reason for not using: https://react.dev/reference/react/useCallback

Reply:
In my opinion it will not make a big difference if you write useCallback with calculateTotalCapacity, and call it separately inside functions changing the capacity. As useEffect is doing the same thing whenever a value of solarPanel in the array is changed calaculateTotalCapacity will be called

Reply:
Okay, This is fine for now but remember that useEffect and useCallback have different purposes. This route can result in excessive re-renders once the calculation changes and dependencies change.